### PR TITLE
Fix CI failure to support SciPy 1.8.0

### LINKED ISSUE #6233: Support SciPy 1.8
## [Tasks](https://github.com/cupy/cupy/wiki/Actions-Needed-for-Dependency-Update)

- [x] Read Release Notes.
- [x] Update FlexCI schema/matrix to test against the version. #6365
- [x] For NumPy/SciPy, update requirements in `setup.py`.
  - [x] Update the pinned upper bound version (see #4773) #6365
  - [x] Backport. #6444
- [x] Update documentation
  - [x] Update Requirements section in [Installation Guide](https://docs.cupy.dev/en/latest/install.html). The baseline version must be updated in the development branch. #6365
  - [x] Update `docs/requirements.txt` to use the latest versions. #6365
  - [x] Backport. Do not update the baseline version in backport. #6444
- [x] Update CuPy codebase (if needed) and backport.
  - #6249
  - #6253
  - #6252

### PR DESCRIPTION
**Pull request by takagi**

### CODE REVIEW

**Comment by takagi** [comment]
Let me test scipy==1.8.0rc2

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py [hunk 1]
```diff
@@ -544,15 +544,19 @@ def test_tocsr_copy(self, xp, sp):
...
-    @testing.numpy_cupy_allclose(sp_name='sp', _check_sparse_format=False)
-    def test_dot_scalar(self, xp, sp):
-        m = _make(xp, sp, self.dtype)
-        return m.dot(2.0)
+    @testing.with_requires('scipy>=1.8.0rc1')
+    def test_dot_scalar(self):
+        for xp, sp in ((numpy, scipy.sparse), (cupy, sparse)):
+            m = _make(xp, sp, self.dtype)
+            with pytest.raises(ValueError):
+                m.dot(2.0)
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py:552]: I found the same issue and reported it as scipy/scipy#15258.

**Pull request merged**